### ISSUE TITLE
[bug] Symlinks inside updater bundle are not preserved on update

### ISSUE THREAD

**Issue by Brendonovich** [tauri-apps/tauri#7480]
### Describe the bug

While #3933 describes an issue with creating updater bundles that contain symlinks, this is a slightly different problem.
We bundle FFMpeg as a macOS framework inside `Spacedrive.app`, and that framework contains symlinks to files within it.
When the updater extracts the downloaded update bundle, it doesn't preserve said symlinks, instead creating empty files.
This causes the app to instantly crash on launch as what should be symlinks are invaild files.

The top Finder window below is what the updater bundle should extract to. I downloaded the updater bundle from where it was being hosted (Firebase Storage) and extracted it, just to make sure it wasn't being affected by something other than the updater.
The bottom Finder window is the contents of the app bundle after running the updater. What were symlinks have become zero size files.
![image](https://github.com/tauri-apps/tauri/assets/14191578/3c5886bf-844e-4b91-881f-67103e84cc29)

### Reproduction

Will try to make a min repro, but it's a bit tricky since it requires having access to an update server

### Platform and versions

```text
[✔] Environment
    - OS: Mac OS 12.5.0 X64
    ✔ Xcode Command Line Tools: installed
    ✔ rustc: 1.70.0 (90c541806 2023-05-31)
    ✔ Cargo: 1.70.0 (ec8a8a0ca 2023-04-25)
    ✔ rustup: 1.25.1 (2022-07-12)
    ✔ Rust toolchain: stable-aarch64-apple-darwin (default)
    - node: 18.12.1
    - pnpm: 8.6.1
    - yarn: 1.22.19
    - npm: 8.19.2

[-] Packages
    - tauri [RUST]: 1.3.0
    - tauri-build [RUST]: 1.3.0
    - wry [RUST]: 0.24.3
    - tao [RUST]: 0.16.2
    - @tauri-apps/api [NPM]: not installed!
    - @tauri-apps/cli [NPM]: 1.3.0 (outdated, latest: 1.4.0)

[-] App
    - build-type: bundle
    - CSP: default-src spacedrive: webkit-pdfjs-viewer: asset: https://asset.localhost blob: data: filesystem: ws: wss: http: https: tauri: 'unsafe-eval' 'unsafe-inline' 'self' img-src: 'self'
    - distDir: ../dist
    - devPath: http://localhost:8001/
    - framework: React
    - bundler: Vite
```

**Comment by lucasfernog** [comment]
I believe #3934 fixed this. It's been mentioned #7773 is also related to this, which we'll merge for 1.5.

**Comment by Brendonovich** [comment]
> I believe #3934 fixed this. It's been mentioned #7773 is also related to this, which we'll merge for 1.5.

@lucasfernog I did test with only #3934 and it didn't fix it, but #7773 did
